app/src/main/java/org/greenstand/android/TreeTracker/permissions/Permissions.kt

```diff
@@ -0,0 +1,211 @@
+                if (event == Lifecycle.Event.ON_START) {
+                    permissionsState.launchMultiplePermissionRequest()
+                }
+            }
+            lifecycleOwner.lifecycle.addObserver(observer)
+            onDispose {
+                lifecycleOwner.lifecycle.removeObserver(observer)
+            }
+        }
+    )
+
+    permissionsState.permissions.forEach { perm ->
+        when (perm.permission) {
+            Manifest.permission.CAMERA -> {
+                when {
+                    perm.hasPermission -> {
+                        return
+                    }
+                    perm.shouldShowRationale -> {
+                        AlertDialog(
+                            onDismissRequest = { navController.popBackStack() },
+                            title = {
+                                Text(text = stringResource(R.string.accept_camera_permission_header))
+                            },
+                            text = {
+                                Text(
+                                    text = stringResource(R.string.accept_camera_permission_message),
+                                    color = Color.Green
+                                )
+                            },
+                            buttons = {
+                                Button(
+                                    modifier = Modifier.wrapContentSize(),
+                                    onClick = {
+                                        perm.launchPermissionRequest()
+                                        navController.popBackStack()
+                                    }
+                                ) {
+                                    Text(
+                                        text = stringResource(R.string.accept_permission)
+                                    )
+                                }
+                            }
+                        )
+                    }
+                    else -> {
+                             PermissionDeniedPermanentlyDialog(navController)
+                    }
+
+                }
+            }
+            Manifest.permission.ACCESS_FINE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        if(state.isLocationEnabled == false){
+                            enableLocation()
+                        }
+                        return
+                    }
+                    perm.shouldShowRationale -> {
+                        LocationRationaleDialog(navController = navController, perm = perm)
+                    }
+                    else -> {
+                            PermissionDeniedPermanentlyDialog(navController)
+                    }
+                }
+            }
+            Manifest.permission.ACCESS_COARSE_LOCATION -> {
+                when {
+                    perm.hasPermission -> {
+                        if(viewModel.isLocationEnabled() == false){

```

Comment:
There is a lint warning here. No need to check equality. Please update.

Reply:
Right not an efficient way. This has been replaced with the state object.